ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,540 @@
+                role = integration.trunc(m.get("role", ""))
+                span.set_tag_str("request.messages.%d.content" % idx, content)
+                span.set_tag_str("request.messages.%d.role" % idx, role)
+
+            if isinstance(messages, list):
+                for idx, message in enumerate(messages):
+                    set_message_tag(message)
+            else:
+                set_message_tag(messages)
+
+        self._record_request(span, kwargs)
+
+        resp, error = yield
+
+        completions = ""
+        if resp and not kwargs.get("stream"):
+            if "choices" in resp:
+                choices = resp["choices"]
+                completions = choices
+                span.set_tag("response.choices.num", len(choices))
+                for choice in choices:
+                    idx = choice["index"]
+                    span.set_tag_str("response.choices.%d.finish_reason" % idx, choice.get("finish_reason"))
+                    if sample_pc_span and choice.get("message"):
+                        span.set_tag(
+                            "response.choices.%d.message.content" % idx,
+                            integration.trunc(choice.get("message").get("content")),
+                        )
+                        span.set_tag(
+                            "response.choices.%d.message.role" % idx,
+                            integration.trunc(choice.get("message").get("role")),
+                        )
+            span.set_tag("response.object", resp["object"])
+            integration.record_usage(span, resp.get("usage"))
+
+            if integration.is_pc_sampled_log(span):
+                messages = kwargs.get("messages")
+                integration.log(
+                    span,
+                    "info" if error is None else "error",
+                    "sampled chat completion",
+                    attrs={
+                        "messages": messages,
+                        "completion": completions,
+                    },
+                )
+
+
+class _EmbeddingHook(_EndpointHook):
+    def handle_request(self, integration, span, args, kwargs):
+        for kw_attr in ["model", "input", "user"]:
+            if kw_attr in kwargs:
+                if kw_attr == "input" and isinstance(kwargs["input"], list):
+                    for idx, inp in enumerate(kwargs["input"]):
+                        span.set_tag_str("request.input.%d" % idx, integration.trunc(inp))
+                span.set_tag("request.%s" % kw_attr, kwargs[kw_attr])
+
+        resp, error = yield
+
+        if resp:
+            if "data" in resp:
+                span.set_tag("response.data.num-embeddings", len(resp["data"]))
+                span.set_tag("response.data.embedding-length", len(resp["data"][0]["embedding"]))
+            for kw_attr in ["object", "usage"]:
+                if kw_attr in kwargs:
+                    span.set_tag("response.%s" % kw_attr, kwargs[kw_attr])
```

Comment:
```suggestion
            if "object" in kwargs:
                span.set_tag("response.object" % kw_attr, kwargs["object"])
```

The usage data should get set by `record_usage` anyways, right?